src/App.tsx

```diff
@@ -15,47 +15,131 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+// Use this function in the render method to prepare goods
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  // To avoid the original array mutation
+  const visibleGoods = [...goods];
+
+  // Sort and reverse goods if needed
+  // eslint-disable-next-line no-console
+  console.log(sortType, isReversed);
+  // sortowanie
+  if (sortType === SortType.ALPHABET) {
+    visibleGoods.sort();
+  }
+
+  if (sortType === SortType.LENGTH) {
+    visibleGoods.sort((a, b) => a.length - b.length);
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
+// DON'T save goods to the state
+// type State = {
+//   isReversed: boolean,
+//   sortType: SortType,
+// };
+
 export const App: React.FC = () => {
+  const [reorderOptions, setReorderOptions] = useState<ReorderOptions>({
+    sortType: SortType.NONE,
+    isReversed: false,
+  });
+
+  const sortAlphabetically = () => {
+    setReorderOptions({
+      sortType: SortType.ALPHABET,
+      isReversed: false,
+    });
+  };
+
+  const sortByLength = () => {
+    setReorderOptions({
+      sortType: SortType.LENGTH,
+      isReversed: false,
+    });
+  };
```

Comment:
These two functions suggest that maybe we might spilt `reorderOptions` state into two separate things.
At the end, in this two functions we do not change `isReserved` and in the `reverseOrder` we do not change `sortType`